### ISSUE TITLE
When GUI slows down, checking for errors for each character is bad

### ISSUE THREAD

**Issue by Seanny123** [nengo/nengo_gui#444]
One of the students of the summer school had to deal with very annoying delays because the server would check for errors for each character they typed. Alternatively, could it wait for the user to stop modifying the text file before sending updates? At least for several seconds?

**Comment by tcstewar** [comment]
> Alternatively, could it wait for the user to stop modifying the text file before sending updates? At least for several seconds?

Good idea.  Even doing something like only doing the update ~2 seconds after the last text change...

**Comment by tcstewar** [comment]
This problem can now be worked-around by toggling auto-updating (thanks to #676).  However, it would be good to do something along these line anyway in the future.

**Comment by krisfrenette** [comment]
This could likely be solved with some sort of debounce on the editor. Something that we can look at in the refactoring,.
